tests/performance/event_generator/tests/test_logeventgenerator.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2024, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+"""Unit tests for the LogEventGenerator class."""
+
+from pathlib import Path
+
+import pytest
+
+from event_generator import LogEventGenerator
+
+
+@pytest.fixture
+def setup_log_generator(tmp_path: Path) -> tuple[LogEventGenerator, Path]:
+    """Setup for LogEventGenerator with a temporary directory.
+
+    Args:
+        tmp_path (LocalPath): Provides a temporary directory path fixture from pytest.
+
+    Returns:
+        tuple: A tuple containing an instance of LogEventGenerator and the path to the log file.
+    """
+    path = tmp_path / "test.log"
+    generator = LogEventGenerator(rate=1, path=str(
+        path), operations=5, max_file_size=1, template_path=None)
+    return generator, path
+
+
+def test_log_creation(setup_log_generator: tuple[LogEventGenerator, Path]) -> None:
+    """Test if log files are created correctly.
+
+    Args:
+        setup_log_generator (fixture): Fixture that provides a log generator and a path.
+    """
+    generator, path = setup_log_generator
+    generator.start()
+    assert path.exists(), "Log file should exist after generation start"
+
+
+def test_log_content(setup_log_generator: tuple[LogEventGenerator, Path]) -> None:
+    """Test the content of log files to ensure logs are written.
+
+    Args:
+        setup_log_generator (fixture): Fixture that provides a log generator and a path.
+    """
+    generator, path = setup_log_generator
+    generator.start()
+    with open(path) as file:
+        content = file.read()
+    assert "This is a test log message" in content, "Log message should be in the log file"
```

Comment:
IMO this is a repetitive test. It should be a test that runs the LogEventGenerator as a fixture. Them three different tests should check if it has been launched in the expected time, if the number of events is correct, and if the file exists. Parametrizing this test for different rates and options could do the job here

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/160e07f8ac2a59a71fdc20cbe4bedc972f51dfec